main.py

```diff
@@ -1,15 +1,26 @@
 # Password Generator
 import random
+import string 
 
+print('@@@@@@@@@@ Welcome to Wales password generator @@@@@@@@@@')
+
+# request user data
+length = int(input('\nEnter the length of password, we recommend 8 chars or more: '))
 first_name = input(str("First Name:"))
 last_name = input(str("Last Name:"))
-num = random.randint(1,500000)
 
-def password_generator(first_name, last_name, number):
```

Comment:
we could also drop this new code into a function to make it more fancy as part of version 0.2, but only if you like the idea of using a function for practice. For practicality, we could just keep it as a plain old script...which technically would be the "right" way since it's simple. But, that won't help you exercise the concept of using functions, so that's a trade off.